appserver/security/core-ee/src/main/java/com/sun/enterprise/security/web/integration/WebPrincipal.java

```diff
@@ -58,8 +59,8 @@
 
     private char[] password;
     private X509Certificate[] certificates;
-    private boolean useCertificate;
-    private SecurityContext securityContext;
+    private final boolean useCertificate;
+    private final SecurityContext securityContext;
```

Comment:
They are not constants! They are only final, not static final. They don't change in the life of the instance, but each instance has it's own field ;)

Reply:
I don't think we have a naming rule in sonar for class level private final fields, we do for public final fields and for private final static fields, so technically David is right 😄. Confusingly, we do have a naming rule for local final fields (https://sonarcloud.io/organizations/payara/rules?languages=java&open=squid%3AS4174&q=final&tags=convention)

Reply:
Yeah no David's point makes sense I'm happy for it to stay as is ( ͡~ ͜ʖ ͡°)

Reply:
@AlanRoth Interesting rule, I'm sure not standard and it may start great confusion, perhaps that's why it is not active in any profile :D
Nice example is here: https://softwareengineering.stackexchange.com/questions/252243/naming-convention-final-fields-not-static